### PR TITLE
new resource: azurerm_search_shared_private_link_service

### PR DESCRIPTION
**Pull request by xiaxyi**
acc tests:
```
--- PASS: TestAccSearchSharedPrivateLinkServiceResource_requiresImport (570.45s)
--- PASS: TestAccSearchSharedPrivateLinkServiceResource_basic (3015.56s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/search        3016.295s

```

### CODE REVIEW

File: internal/services/search/search_shared_private_link_service_resource.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+			ValidateFunc: azure.ValidateResourceID,
+		},
+
+		"request_message": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_region": {
```
Comment (tombuildsstuff) [inline internal/services/search/search_shared_private_link_service_resource.go:72]: we use `location` rather than region - should this also be normalized?

**Comment by xiaxyi** [comment]
Thanks @tombuildsstuff  Let me confirm with the service team about more backgrounds of this property, will update soon

**Comment by xiaxyi** [comment]
hi @tombuildsstuff , I removed the property `location` after confirmed with service team. This property is not used now, they introduced it for future usage.

**Review by katbyte — APPROVED** [review]
LGTM 🧾

**Pull request merged**